File: src/main/java/at/jku/ssw/coco/intellij/symbolcontributor/CocoChooseByNameProductionContributor.java
package at.jku.ssw.coco.intellij.symbolcontributor;

import at.jku.ssw.coco.intellij.CocoUtil;
import at.jku.ssw.coco.intellij.psi.CocoProduction;
import com.intellij.navigation.ChooseByNameContributor;
import com.intellij.navigation.NavigationItem;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Thomas on 29/03/2015.
 */
public class CocoChooseByNameProductionContributor implements ChooseByNameContributor {
    @NotNull
    @Override
    public String[] getNames(Project project, boolean includeNonProjectItems) {
        return CocoUtil.findProductions(project)
                .parallelStream()
                .map(CocoProduction::getName)
                .toArray(String[]::new);
    }

    @NotNull
    @Override
    public NavigationItem[] getItemsByName(String name, String pattern, Project project, boolean includeNonProjectItems) {
        List<CocoProduction> productions = CocoUtil.findProductions(project, name);
        return productions.toArray(new NavigationItem[productions.size()]);
    }
}

File: src/main/kotlin/me/salzinger/intellij/coco/symbolcontributor/CocoChooseByNameProductionContributor.kt
package me.salzinger.intellij.coco.symbolcontributor

import com.intellij.navigation.ChooseByNameContributor
import com.intellij.navigation.NavigationItem
import com.intellij.openapi.project.Project
import me.salzinger.intellij.coco.CocoUtil

/**
 * Created by Thomas on 29/03/2015.
 */
class CocoChooseByNameProductionContributor : ChooseByNameContributor {
    override fun getNames(project: Project, includeNonProjectItems: Boolean): Array<String> {
        return CocoUtil.findProductions(project)
            .mapNotNull { it.name }
            .toTypedArray()
    }

    override fun getItemsByName(
        name: String,
        pattern: String,
        project: Project,
        includeNonProjectItems: Boolean
    ): Array<NavigationItem> {
        return CocoUtil.findProductions(project, name).toTypedArray()
    }
}
